rdwatch/core/views/model_run.py

```diff
@@ -388,30 +357,30 @@ def create_model_run(
     }
 
 
-def _get_model_runs_cache_key(params: dict) -> str:
-    """Generate a cache key for the model runs listing endpoint."""
-    most_recent_evaluation = ModelRun.objects.aggregate(
-        latest_eval=Max('evaluations__timestamp')
-    )['latest_eval']
-
-    return '|'.join(
-        [
-            ModelRun.__name__,
-            str(most_recent_evaluation),
-            *[f'{key}={value}' for key, value in params.items()],
-        ]
-    ).replace(' ', '_')
-
-
 @router.get('/', response={200: list[ModelRunListSchema]})
 @paginate(ModelRunPagination, page_size=MODEL_RUN_PAGE_SIZE)
 def list_model_runs(
     request: HttpRequest,
     filters: ModelRunFilterSchema = Query(...),  # noqa: B008
 ):
+    # TODO maybe the aggregate stats should have a separate endpoint since they only need to
+    # be queried when the filters change, not every time a new page is loaded
     return filters.filter(get_queryset())
 
 
+@router.post(
+    '/{id}/finalization/',
+    response={200: ModelRunDetailSchema},
+    auth=[ModelRunAuth()],
+)
+# this is safe because we're using a nonstandard header w/ API Key for auth
+@csrf_exempt
+def finalize_model_run(request: HttpRequest, id: UUID4):
+    model_run = get_object_or_404(ModelRun, pk=id)
+    model_run.compute_aggregate_stats()
+    return model_run
```

Comment:
`ModelRunDetailSchema` includes some additional fields not on `ModelRun`. When we use it in the `create_model_run` endpoint above, we explicitly return a dictionary that is the expected structure, like this -
```suggestion
@router.post(
    '/{id}/finalization/',
    response={200: ModelRunDetailSchema},
    auth=[ModelRunAuth()],
)
# this is safe because we're using a nonstandard header w/ API Key for auth
@csrf_exempt
def finalize_model_run(request: HttpRequest, id: UUID4):
    model_run = get_object_or_404(ModelRun, pk=id)
    model_run.compute_aggregate_stats()
    return {
        'id': model_run.pk,
        'title': model_run.title,
        'performer': {
            'id': model_run.performer.pk,
            'team_name': model_run.performer.team_name,
            'short_code': model_run.performer.short_code,
        },
        'region_name': model_run.region.name,
        'parameters': model_run.parameters,
        'numsites': 0,
        'created': model_run.created,
        'expiration_time': model_run.expiration_time,
    }
```

Currently, the `/finalize/` endpoint always returns a 500 error because of this.

Reply:
Ah good catch, thanks